client/my-sites/site-settings/redirect-non-jetpack/index.jsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpack = redirectRoute => WrappedComponent => {
+	class RedirectNonJetpack extends Component {
+		static propTypes = {
+			// Connected props
+			siteIsAtomic: PropTypes.bool,
+			siteIsJetpack: PropTypes.bool,
+			siteSlug: PropTypes.string
+		}
+
+		componentDidMount() {
+			this.redirectIfNoAccess();
+		}
+
+		componentDidUpdate() {
+			this.redirectIfNoAccess();
+		}
+
+		redirectIfNoAccess() {
+			if ( this.props.siteIsJetpack === false || this.props.siteIsAtomic ) {
+				this.redirect();
+			}
+		}
+
+		redirect = () => {
+			const { siteSlug } = this.props;
+
+			if ( redirectRoute ) {
+				return page( redirectRoute );
+			}
+
+			if ( siteSlug ) {
+				page( '/settings/general/' + siteSlug );
+			}
+		};
+
+		render() {
+			return (
+				<WrappedComponent
+					redirect={ this.redirect }
+					{ ...this.props }
+				/>
+			);
+		}
+	}
+	const connectComponent = connect(
+		( state ) => {
+			const siteId = getSelectedSiteId( state );
+
+			return {
+				siteIsAtomic: isSiteAutomatedTransfer( state, siteId ),
+				siteIsJetpack: isJetpackSite( state, siteId ),
+				siteSlug: getSelectedSiteSlug( state ),
+			};
+		}
+	);
+
+	return flowRight(
+		connectComponent,
+		localize
```

Comment:
Does this HOC really need `localize` at all? I don't see it used.

Reply:
Ah, not anymore after the refactors I guess. @AnnaMag wanna do a quick PR to remove it?

Reply:
Thanks for spotting it @jsnajdr and @tyxla for landing the code.
I tried to do a PR on master and the code is not part of it yet. Should I wait a bit or do a PR against this branch? Thanks!

Reply:
@AnnaMag it has been merged in master already; make sure to pull the latest one and create your branch against it.